transcom/mymove
# [MB-16227] Set a nil destination address for an HHG to new duty station location address

## [Jira ticket](https://dp3.atlassian.net/browse/MB-16227)

## Summary

In this PR, we are changing how an HHG shipment's destination address is handled when it is not known (i.e., is `nil` in the payload).  For HHG shipments only, we now make a copy of the city/state/zip/country of the new duty location address and set that to be the destination address (we also set street address 1 to `N/A` because the Address model requires a non-empty string).  This will apply whether it's the service member, the office user, or the prime creating the HHG shipment.  We do this because we need a zip available for pricing certain service items which may be completed prior to having a final destination address ([more context here](https://dp3.atlassian.net/wiki/spaces/MT/pages/2158198983/Options+for+MB-15945+Populate+destination+address+with+destination+duty+location+if+unknown)).

Note that we are only able to do this on shipment creation at the moment and not on shipment update due to [this existing issue](https://dp3.atlassian.net/browse/MB-15718) around "unsetting" a destination address.  Once that issue is fixed, a similar default can be done on update when resetting the destination address to nil.

This will only affect new shipments created going forward after this lands.  There is a [separate PR](https://github.com/transcom/mymove/pull/10866) with a migration for adjusting existing HHG shipments to reflect this new behavior -- it will need to land after this PR.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

Try creating an HHG shipment as a service member, a service counselor, and a prime.  Verify that a nil destination address defaults to the new duty location address (in the UI or the database if you have access).  Other shipment types should not do this default.

You can also try running one of your HHG shipments with a nil destination address all the way through to the payment request stage and verify that everything seems to work as expected.

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
